lxd/storage/drivers/driver_alletra.go

```diff
@@ -0,0 +1,279 @@
+	oldMode := d.config["alletra.mode"]
+
+	// Ensure alletra.mode cannot be changed to avoid leaving volume mappings
+	// and prevent disturbing running instances.
+	if oldMode != "" && oldMode != newMode {
+		return errors.New("HPE Alletra Storage mode cannot be changed")
+	}
+
+	// Check if the selected HPE Alletra Storage mode is supported on this node.
+	// Also when forming the storage pool on a LXD cluster, the mode
+	// that got discovered on the creating machine needs to be validated
+	// on the other cluster members too. This can be done here since Validate
+	// gets executed on every cluster member when receiving the cluster
+	// notification to finally create the pool.
+	if newMode != "" {
+		connector, err := connectors.NewConnector(newMode, "")
+		if err != nil {
+			return fmt.Errorf("HPE Alletra Storage mode %q is not supported: %w", newMode, err)
+		}
+
+		err = connector.LoadModules()
+		if err != nil {
+			return fmt.Errorf("HPE Alletra Storage mode %q is not supported due to missing kernel modules: %w", newMode, err)
+		}
+	}
+
+	return nil
+}
+
+// Create is called during pool creation and is effectively using an empty driver struct.
+// WARNING: The Create() function cannot rely on any of the struct attributes being set.
+func (d *alletra) Create() error {
+	err := d.FillConfig()
+	if err != nil {
+		return err
+	}
+
+	// Validate both pool and gateway here and return an error if they are not set.
+	// Since those aren't any cluster member specific keys the general validation
+	// rules allow empty strings in order to create the pending storage pools.
+	if d.config["alletra.wsapi.url"] == "" {
+		return errors.New("The alletra.wsapi.url cannot be empty")
+	}
+
+	if d.config["alletra.wsapi.user.name"] == "" {
+		return errors.New("The alletra.wsapi.user.name cannot be empty")
+	}
+
+	if d.config["alletra.wsapi.user.password"] == "" {
+		return errors.New("The alletra.wsapi.user.password cannot be empty")
+	}
+
+	err = d.client().createVolumeSet()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Delete removes a storage pool.
+func (d *alletra) Delete(op *operations.Operation) error {
+	err := d.client().deleteVolumeSet()
+	if err != nil {
+		return err
+	}
+
+	// If the user completely destroyed it, call it done.
+	if !shared.PathExists(GetPoolMountPath(d.name)) {
+		return nil
+	}
+
+	// On delete, wipe everything in the directory.
+	return wipeDirectory(GetPoolMountPath(d.name))
+}
+
+// Update applies any driver changes required from a configuration change.
+func (d *alletra) Update(changedConfig map[string]string) error {
+	return nil
+}
+
+// Mount mounts the storage pool.
+func (d *alletra) Mount() (bool, error) {
+	return true, nil
+}
+
+// Unmount unmounts the storage pool.
+func (d *alletra) Unmount() (bool, error) {
+	return true, nil
+}
+
+// GetResources returns the pool resource usage information.
+func (d *alletra) GetResources() (*api.ResourcesStoragePool, error) {
+	return nil, nil
```

Comment:
@mihalicyn this to be implemented in a future PR? Should probably return error unsupported for now.

Reply:
> this to be implemented in a future PR?

Nope, because there is no clean or good way to obtain such an information. There is a way to get some very-very low level details about disk sizes and CPGs, but nothing to get that statistics for a volume set.

This is not going to be implemented.